smtp_connector.py

```diff
@@ -103,6 +96,23 @@ def initialize(self):
 
         return phantom.APP_SUCCESS
 
+    def _get_auth_type(self):
+
+        config = self.get_config()
+        username = config.get('username')
+        client_id = config.get('client_id', '')
+        client_secret = config.get('client_secret', '')
+
+        password = config.get('password')
+
+        if username:
+            if client_id and client_secret:
```

Comment:
should we fail if they specify one of these and not the other?

Reply:
We are not failing it as we are defined fixed the priority of the authentication in the following in order from the provided inputs
OAuth >>  Basic Auth >> Authless

Therefore if enough data is not provided for OAuth it will try with Basic and at last Authless